tools/java/org/unicode/cldr/util/VettingViewer.java

```diff
@@ -805,6 +807,161 @@ public void generateSummaryHtmlErrorTables(Appendable output, EnumSet<Choice> ch
+        private VettingViewer<T>.FileInfo totals;
+        private Map<String, VettingViewer<T>.FileInfo> localeNameToFileInfo;
+        private String header;
+        private int configParallel; // parallelism. 0 means "let Java decide"
+        private int configChunkSize; // Number of locales to process at once, minimum 1
+
+        @SuppressWarnings("unchecked")
+        public WriteContext(Set<Entry<String, String>> entrySet, EnumSet<Choice> choices, T organization, FileInfo totals, Map<String, VettingViewer<T>.FileInfo> localeNameToFileInfo, String header) {
+            for(Entry<String, String> e : entrySet) {
+                names.add(e.getKey());
+                locales.add(e.getValue());
+            }
+            int count = names.size();
+            this.outputs = new StringBuffer[count];
+            for(int i=0;i<count;i++) {
+                outputs[i] = new StringBuffer();
+            }
+            if(DEBUG_THREADS) System.err.println("Initted " + this.outputs.length + " outputs");
+
+            // other data
+            this.choices = choices;
+
+            thingsThatRequireOldFile = EnumSet.of(Choice.englishChanged, Choice.missingCoverage, Choice.changedOldValue);
+            ourChoicesThatRequireOldFile = choices.clone();
+            ourChoicesThatRequireOldFile.retainAll(thingsThatRequireOldFile);
+
+            this.organization = organization;
+            this.totals = totals;
+            this.localeNameToFileInfo = localeNameToFileInfo;
+            this.header = header;
+
+            if(DEBUG_THREADS) System.err.println("writeContext for " + organization.toString() + " booted with " + count + " locales");
+
+            // setup env
+            CLDRConfig config = CLDRConfig.getInstance();
+
+            this.configParallel = Math.max(config.get("CLDR_VETTINGVIEWER_PARALLEL", 0), 0);
+            if(this.configParallel < 1) {
+                this.configParallel = java.lang.Runtime.getRuntime().availableProcessors(); // matches ForkJoinPool() behavior
+            }
+            this.configChunkSize = Math.max(config.get("CLDR_VETTINGVIEWER_CHUNKSIZE", 1), 1);
+            System.err.println("vv: CLDR_VETTINGVIEWER_PARALLEL="+configParallel+", CLDR_VETTINGVIEWER_CHUNKSIZE="+configChunkSize);
+        }
+
+        public void append(Appendable output) throws IOException {
+            // all done, append all
+            char lastChar = ' ';
+
+            for(int n=0;n<outputs.length;n++) {
+              final String name = names.get(n);
+              if(DEBUG_THREADS) System.err.println("Appending " + name + " - " + outputs[n].length());
+              output.append(outputs[n]);
+
+              char nextChar = name.charAt(0);
+              if (lastChar != nextChar) {
+                  output.append(this.header);
+                  lastChar = nextChar;
+              }
+            }
+        }
+
+        public int size() {
+            return names.size();
+        }
+
```

Comment:
It seems like it would be simpler and more understandable to use parallel streams. Or is there a problem with that?

Reply:
They are parallel streams? And then they are appended together at the end, including header-inclusion logic.

Reply:
streams outputs[0…n] are written in parallel by separate threads, using the ForkJoin mechanism's algorithms for efficient scheduling.  Then, all of the streams are written one after another to the requested output.

Reply:
Here is a tutorial on them. https://docs.oracle.com/javase/tutorial/collections/streams/parallelism.html

There are a few instances in the unitTests, but we haven't changed older code.

Reply:
OK, that's a different use of 'parallel stream' but, I think it's worth looking into.